src/models/session/mocks/complex-flow-breadcrumbs.ts

```diff
@@ -600,96 +613,93 @@ export const normalizedFlow: NormalizedFlow = [
       text: "contains",
     },
     type: ComponentType.Question,
-    parentId: "HV0gV8DOil",
-    rootNodeId: "2PT6bTPTqj",
-    edges: ["oB2vfxQs4D", "ykhO0drpaY", "U9S73zxy9n", "LwozLZdXCA"],
-  },
-  {
-    id: "oB2vfxQs4D",
-    data: {
-      val: "food.fruit.apple",
-      text: "apples",
+    children: {
+      oneOf: [
+        {
+          id: "oB2vfxQs4D",
+          data: {
+            val: "food.fruit.apple",
+            text: "apples",
+          },
+          type: ComponentType.Answer,
+          children: {
+            eachOf: [
+              {
+                id: "6RR1J1lmrM",
+                data: {
+                  color: "#EFEFEF",
+                  title: "🍏",
+                  resetButton: false,
+                },
+                type: ComponentType.Notice,
+              },
+            ],
+          },
+        },
+        {
+          id: "ykhO0drpaY",
+          data: {
+            val: "food.bread",
+            text: "bread",
+          },
+          type: ComponentType.Answer,
+          children: {
+            eachOf: [
+              {
+                id: "g0IAKsBVPQ",
+                data: {
+                  color: "#EFEFEF",
+                  title: "🥖",
+                  resetButton: false,
+                },
+                type: ComponentType.Notice,
+              },
+            ],
+          },
+        },
+        {
+          id: "U9S73zxy9n",
+          data: {
+            val: "food.fruit.apple,food.bread",
+            text: "apples and bread",
+          },
+          type: ComponentType.Answer,
+          children: {
+            eachOf: [
+              {
+                id: "t3SCqQKeUK",
+                data: {
+                  color: "#EFEFEF",
+                  title: "🍏🥖",
+                  resetButton: false,
+                },
+                type: ComponentType.Notice,
+              },
+            ],
+          },
+        },
+        {
+          id: "LwozLZdXCA",
+          data: {
+            text: "neither apples nor bread",
+          },
+          type: ComponentType.Answer,
+          children: {
+            eachOf: [
+              {
+                id: "52ZNXBMLDP",
+                data: {
+                  color: "#EFEFEF",
+                  title: "?, so must be a 🍌 or 🔧",
+                  resetButton: false,
+                },
+                type: ComponentType.Notice,
+              },
+            ],
+          },
+        },
+      ],
```

Comment:
Hopefully, this data structure is intuitive and self-explanatory. You can see how the structure of the flow is encoded as a data structure which could easily be traversed.